### PR TITLE
Avoid exporting superfluous meters and keys in staff groups

### LINKED ISSUE #1129: MusicXML importer: PartStaffs end up with too many (2) TimeSignatures in a non-first Measure
**music21 version**

`7.1`   

**Problem summary**

Attached is a MusicXML file that music21 wrote from a stream containing two PartStaffs (and a StaffGroup relating them).  It's an Organ score, with 5 measures, each of which has a new time signature: 7/8, 8/8, 7/8, 8/8, and 11/16, respectively.  I believe the MusicXML is correct, but I might be wrong.  It displays correctly in any case.  When I import this MusicXML file back into music21, however, while the first measure's time signature looks right, the second measure's time signature (and I'm guessing the others as well) shows up in the measure twice (in each PartStaff)

If I export a different MusicXML file from an otherwise identical Score containing Parts instead of PartStaffs, and then import _that_, all is well (I get one time signature at the beginning of each Measure in each Part).

I would note that neither MusicXML file appears to have the extra time signatures in it, so I'm guessing this is a bug in MusicXML import, that only happens when a part has more than one staff.

I have attached both MusicXML files for your enjoyment (zipped).

[LesCorps_Parts.musicxml.zip](https://github.com/cuthbertLab/music21/files/7173870/LesCorps_Parts.musicxml.zip)
[LesCorps_PartStaffs.musicxml.zip](https://github.com/cuthbertLab/music21/files/7173871/LesCorps_PartStaffs.musicxml.zip)

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Take 2 -- fixes #1129. Alternative to #1131.

@gregchapman-dev Another one here -- would you mind checking?

### CODE REVIEW

**Comment by gregchapman-dev** [comment]
Code looks good by inspection... I'm not in a position to test this directly at the moment (everything is pulled apart), but your test case looks like it tests what I would have tested.  Thanks!

File: music21/musicxml/m21ToXml.py [hunk 1]
```diff
@@ -2469,6 +2471,8 @@ def __init__(self, partObj: Union[stream.Part, stream.Score, None] = None, paren
             self.midiChannelList = parent.midiChannelList  # shared list
             self.makeNotation = parent.makeNotation
 
+        self.previous_sibling_in_group: Optional[stream.PartStaff] = None
```
Comment (mscuthbert) [inline music21/musicxml/m21ToXml.py:2474]: since this is a public interface it should be camelCased.

Comment (mscuthbert) [reply music21/musicxml/m21ToXml.py:2474]: Explaining what a sibling is would be helpful.

File: music21/musicxml/m21ToXml.py [hunk 2]
```diff
@@ -5654,6 +5660,32 @@ def wrapObjectInAttributes(self, objectToWrap, methodToMx):
 
         return mxAttributes
 
+    def _matches_previous_sibling_in_group(self,
```
Comment (mscuthbert) [inline music21/musicxml/m21ToXml.py:5663]: this is okay as underscore case as private, but the name should match the attribute.

File: music21/musicxml/m21ToXml.py [hunk 3]
```diff
@@ -5977,20 +6009,21 @@ def setMxAttributesObjectForStartOfMeasure(self):
...
-            smts = list(m.getElementsByClass('SenzaMisuraTimeSignature'))
-            if smts:
-                mxAttributes.append(self.timeSignatureToXml(smts[0]))
-
-            # For staves, see joinPartStaffs()
-            # TODO: part-symbol
-            # TODO: instruments
-            if m.clef is not None:
-                mxAttributes.append(self.clefToXml(m.clef))
+        if m.keySignature is not None and not self._matches_previous_sibling_in_group(
```
Comment (mscuthbert) [inline music21/musicxml/m21ToXml.py:6012]: break two-part clauses between "None / and" -- wrap in parentheses.  (same below)

**Review by mscuthbert — CHANGES_REQUESTED** [review]
LGTM -- just two style changes I'd prefer.  Thanks!

File: music21/musicxml/m21ToXml.py (hunk 1 above)
Comment (jacobtylerwalls) [reply music21/musicxml/m21ToXml.py:2474]: Yeah, and it's not really any more keystrokes to call it `previousPartStaffInGroup` either...

**Pull request merged**